Error: Input required: backendAzureRmResourceGroupLocation

**Describe the bug**
The task seem to require `backendAzureRmResourceGroupLocation` input after latest release? Since we cannot control the version of individual tasks & extensions, all our pipelines are broken. 

**To Reproduce**
```
  - task: TerraformCLI@0
    displayName: 'Terraform init'
    inputs:
      command: 'init'
      workingDirectory: '${{ parameters.workingDirectory }}'
      backendType: 'azurerm'
      backendServiceArm: '${{ parameters.subscription }}'
      backendAzureRmResourceGroupName: '%rg-name%'
      backendAzureRmStorageAccountName: '%account-name%'
      backendAzureRmContainerName: '%container-name%'
      backendAzureRmKey: '${{ parameters.backendKey }}'
      allowTelemetryCollection: false
```

**Expected behavior**
I would expect the task to work as before without requiring `backendAzureRmResourceGroupLocation` input

**Pipeline Logs**
```
Starting: Terraform init
==============================================================================
Task         : Terraform CLI
Description  : Execute terraform cli commands
Version      : 0.7.0
Author       : Charles Zipp
Help         : 
==============================================================================
C:\hostedtoolcache\windows\terraform\1.0.1\x64\terraform.exe version
Terraform v1.0.1
on windows_amd64

Your version of Terraform is out of date! The latest version
is 1.1.2. You can update by downloading from https://www.terraform.io/downloads.html
C:\hostedtoolcache\windows\terraform\1.0.1\x64\terraform.exe init -backend-config=storage_account_name=*** -backend-config=container_name=*** -backend-config=key=*** -backend-config=resource_group_name=*** -backend-config=subscription_id=*** -backend-config=tenant_id=*** -backend-config=client_id=*** -backend-config=client_secret=***

Initializing the backend...

Successfully configured the backend "azurerm"! Terraform will automatically
use this backend unless the backend configuration changes.

Initializing provider plugins...
- Finding latest version of hashicorp/random...
- Finding hashicorp/azurerm versions matching "2.65.0"...
- Finding hashicorp/azuread versions matching "1.4.0"...
- Installing hashicorp/random v3.1.0...
- Installed hashicorp/random v3.1.0 (signed by HashiCorp)
- Installing hashicorp/azurerm v2.65.0...
- Installed hashicorp/azurerm v2.65.0 (signed by HashiCorp)
- Installing hashicorp/azuread v1.4.0...
- Installed hashicorp/azuread v1.4.0 (signed by HashiCorp)

Terraform has created a lock file .terraform.lock.hcl to record the provider
selections it made above. Include this file in your version control repository
so that Terraform can guarantee to make the same selections by default when
you run "terraform init" in the future.

Terraform has been successfully initialized!
##[error]Error: Input required: backendAzureRmResourceGroupLocation
Finishing: Terraform init
```

**Agent Configuration**
 - OS: both Ubuntu 20.04 & Windows Server 2019
 - both Hosted & Self Hosted
 - Terraform version used: 1.0.1
 - AzureCLI version used: N/A

**Additional context**
Add any other context about the problem here.

--- a/pipelines/pipeline-beta.yml
+++ b/pipelines/pipeline-beta.yml
@@ -55,4 +55,5 @@ stages:
         - init_with_self_configured_backend.yml
         - switch_workspaces.yml  
         - aws_self_configured.yml  
-        - aws_service_connection.yml 
+        - aws_service_connection.yml
+        - init_without_location.yml

--- a/pipelines/pipeline-rc.yml
+++ b/pipelines/pipeline-rc.yml
@@ -57,3 +57,4 @@ stages:
         - switch_workspaces.yml
         - aws_self_configured.yml
         - aws_service_connection.yml
+        - init_without_location.yml

--- a/pipelines/test/init_without_location.yml
+++ b/pipelines/test/init_without_location.yml
@@ -1,0 +1,29 @@
+parameters:
+  stage: ''
+
+jobs:
+- job: init_without_location${{ parameters.stage }}
+  variables: 
+    test_templates_dir: $(terraform_templates_dir)/version11
+  steps:
+    - task: DownloadPipelineArtifact@2
+      displayName: download terraform templates
+      inputs: 
+        artifact: terraform_templates
+        path: $(terraform_extension_dir)
+    - task: charleszipp.azure-pipelines-tasks-terraform-${{ parameters.stage }}.azure-pipelines-tasks-terraform-installer.TerraformInstaller@0
+      displayName: install terraform
+      inputs:
+        terraformVersion: 1.0.10
+    - task: charleszipp.azure-pipelines-tasks-terraform-${{ parameters.stage }}.azure-pipelines-tasks-terraform-cli.TerraformCLI@0
+      displayName: 'terraform init'
+      inputs:
+        command: init
+        workingDirectory: $(test_templates_dir)
+        backendType: azurerm
+        backendServiceArm: 'env_test'
+        backendAzureRmResourceGroupName: rg-trfrm-${{ parameters.stage }}-eus-czp
+        backendAzureRmStorageAccountName: sttrfrm${{ parameters.stage }}eusczp
+        backendAzureRmStorageAccountSku: Standard_RAGRS
+        backendAzureRmContainerName: azure-pipelines-terraform
+        backendAzureRmKey: init_without_location.tfstate

--- a/pipelines/test/init_without_location.yml
+++ b/pipelines/test/init_without_location.yml
@@ -2,9 +2,7 @@ parameters:
   stage: ''
 
 jobs:
-- job: init_without_location${{ parameters.stage }}
-  variables: 
-    test_templates_dir: $(terraform_templates_dir)/version11
+- job: init_without_location_${{ parameters.stage }}
   steps:
     - task: DownloadPipelineArtifact@2
       displayName: download terraform templates
@@ -19,7 +17,7 @@ jobs:
       displayName: 'terraform init'
       inputs:
         command: init
-        workingDirectory: $(test_templates_dir)
+        workingDirectory: $(terraform_templates_dir)
         backendType: azurerm
         backendServiceArm: 'env_test'
         backendAzureRmResourceGroupName: rg-trfrm-${{ parameters.stage }}-eus-czp

--- a/tasks/terraform-cli/src/context/azdo-task-context.ts
+++ b/tasks/terraform-cli/src/context/azdo-task-context.ts
@@ -72,7 +72,7 @@ export default class AzdoTaskContext implements ITaskContext {
     get backendAzureRmResourceGroupName() {
         return this.getInput("backendAzureRmResourceGroupName", true);
     }
-    @trackValue("inputs.backends.azurerm.rg.location", usesBackend(BackendTypes.azurerm))
+    @trackValue("inputs.backends.azurerm.rg.location", (ctx: ITaskContext) => isCommand("init") && ctx.backendType == BackendTypes.azurerm && ctx.ensureBackend === true)
     get backendAzureRmResourceGroupLocation() {
         return this.getInput("backendAzureRmResourceGroupLocation", true);
     }

